lib/mix/lib/mix/task.ex

```diff
@@ -445,32 +499,60 @@ defmodule Mix.Task do
     end
   end
 
-  defp run_alias([h | t], alias_args, proj, original_task, _res) when is_binary(h) do
+  defp run_alias([h | t], alias_args, proj, original_task, options, _res) when is_binary(h) do
     case OptionParser.split(h) do
       [^original_task | args] ->
-        res = run_task(proj, original_task, args ++ alias_args)
-        run_alias(t, [], proj, original_task, res)
+        res = run_task(proj, original_task, args ++ alias_args, options)
+        run_alias(t, [], proj, original_task, options, res)
 
       [task | args] ->
-        res = run(task, join_args(args, alias_args, t))
-        run_alias(t, alias_args, proj, original_task, res)
+        res =
+          case options do
+            [apps: apps] ->
+              run_in_apps(task, apps, join_args(args, alias_args, t))
+
+            _ ->
+              run(task, join_args(args, alias_args, t))
+          end
+
+        run_alias(t, alias_args, proj, original_task, options, res)
     end
   end
 
-  defp run_alias([h | t], alias_args, proj, original_task, _res) when is_function(h, 1) do
-    res = h.(join_args([], alias_args, t))
-    run_alias(t, alias_args, proj, original_task, res)
+  defp run_alias([h | t], alias_args, proj, original_task, options, _res)
+       when is_function(h, 1) do
+    res =
+      case options do
+        [apps: apps] ->
+          if Mix.Project.umbrella?() do
```

Comment:
I think the code here is hard to read and follow. I will work on it a little bit.

Reply:
I think if you replace options by `in_apps` you can have a nice `cond`:

```elixir
cond do
  in_apps && Mix.Project.umbrella?() ->

  in_apps ->

  true ->

end
```

Reply:
Thanks for the suggestion. Everything is simpler now with the removal of the options arg 👍